extensions/functions/src/main/java/io/crate/window/OffsetValueFunctions.java

```diff
@@ -242,10 +245,10 @@ public Object execute(int idxInPartition,
         } else {
             offset = 1;
         }
-        if (offset == 0) {
-            ignoreNulls = false;
+        if (offset == 0 && ignoreNulls) {
+            throw new InvalidArgumentException("offset 0 is not a valid argument if ignore nulls flag is set");
```

Comment:
Oracle throws when offset 0 with `ignore nulls` "ORA-01428: argument '0' is out of range"
I think it makes sense because lead(x,0) where the idx is at value null, it is unclear what is the expected behaviour. Returning a default could be an option but the description of default is:
> The optional default value is returned if the offset goes beyond the scope of the table. 

which is not the case.